src/wayland/wl.c

```diff
@@ -332,10 +332,10 @@ void wl_deinit(void) {
         // could have been aborted half way through, or the compositor doesn't
         // support some of these features.
         if (ctx.layer_surface != NULL) {
-                zwlr_layer_surface_v1_destroy(ctx.layer_surface);
+                g_clear_pointer(&ctx.layer_surface, zwlr_layer_surface_v1_destroy);
```

Comment:
Why was this needed?

Reply:
this was taken from the original commit made by friso